core/src/test/java/com/graphhopper/routing/subnetwork/PrepareRoutingSubnetworksTest.java

```diff
@@ -71,60 +84,68 @@ private static GraphHopperStorage createSubnetworkTestStorage(EncodingManager em
 
     @Test
     public void testRemoveSubnetworkIfOnlyOneVehicle() {
-        FlagEncoder encoder = em.fetchEdgeEncoders().iterator().next();
-        GraphHopperStorage g = createSubnetworkTestStorage(em, encoder);
-        PrepareRoutingSubnetworks instance = new PrepareRoutingSubnetworks(g, Collections.singletonList(
-                new PrepareRoutingSubnetworks.PrepareJob("car", accessEnc, NO_TURN_COST_PROVIDER)));
+        GraphHopperStorage g = createSubnetworkTestStorage("car");
+        EncodingManager em = g.getEncodingManager();
+        FlagEncoder encoder = g.getEncodingManager().fetchEdgeEncoders().iterator().next();
+        PrepareRoutingSubnetworks instance = new PrepareRoutingSubnetworks(g, Collections.singletonList(createJob(em, encoder, NO_TURN_COST_PROVIDER)));
         // this rules out the upper small network
         instance.setMinNetworkSize(4);
         instance.doWork();
         assertTrue(GHUtility.getProblems(g).isEmpty());
-        EdgeExplorer explorer = g.createEdgeExplorer(DefaultEdgeFilter.allEdges(encoder.getAccessEnc()));
+        EdgeExplorer explorer = g.createEdgeExplorer(createFilter(em, encoder.toString()));
         assertEquals(GHUtility.asSet(), GHUtility.getNeighbors(explorer.setBaseNode(4)));
 
         // this time we lower the threshold and the small network will remain
-        g = createSubnetworkTestStorage(em, encoder);
-        instance = new PrepareRoutingSubnetworks(g, Collections.singletonList(
-                new PrepareRoutingSubnetworks.PrepareJob("car", accessEnc, NO_TURN_COST_PROVIDER)));
+        g = createSubnetworkTestStorage("car");
+        em = g.getEncodingManager();
+        encoder = g.getEncodingManager().fetchEdgeEncoders().iterator().next();
+        instance = new PrepareRoutingSubnetworks(g, Collections.singletonList(createJob(em, encoder, NO_TURN_COST_PROVIDER)));
         instance.setMinNetworkSize(3);
         instance.doWork();
-        explorer = g.createEdgeExplorer(DefaultEdgeFilter.allEdges(encoder.getAccessEnc()));
+        explorer = g.createEdgeExplorer(createFilter(em, encoder.toString()));
         assertEquals(GHUtility.asSet(5, 6), GHUtility.getNeighbors(explorer.setBaseNode(4)));
     }
 
+    private static EdgeFilter createFilter(EncodingManager em, String str) {
+        BooleanEncodedValue inSubnetwork = em.getBooleanEncodedValue(InSubnetwork.key(str));
+        BooleanEncodedValue accessEnc = em.getEncoder(str).getAccessEnc();
+        return e -> !e.get(inSubnetwork) && e.get(accessEnc);
```

Comment:
Hm, it seems a bit strange this filter comes up here again(?)

Reply:
Why? Otherwise we would ignore the subnetwork EncodedValue when collecting and asserting the neighbors

Reply:
Ah it was only for asserting yes. But simply getting a list of marked subnetwork edges and comparing this list with our expectations seems a lot more intuitive to me. See #2295